dev/vcpkg/setup-build-depends.sh

```diff
@@ -45,21 +45,54 @@ install_gcc9_from_source() {
             make -j$(nproc)
             make install
 
-            update-alternatives --install /usr/bin/gcc gcc /usr/local/${gcc_version}/bin/gcc 900 --slave /usr/bin/g++ g++ /usr/local/${gcc_version}/bin/g++
+            update-alternatives --install /usr/bin/gcc gcc ${gcc_install_dir}/bin/gcc 900 --slave /usr/bin/g++ g++ ${gcc_install_dir}/bin/g++
+            echo ${gcc_install_dir}/lib64 > /etc/ld.so.conf.d/gcc.conf && ldconfig
      fi
 }
 
+install_binutils_from_source() {
+    binutils_version=binutils-2.41
+    binutils_install_dir=/usr/local/${binutils_version}
+    cd /tmp
+    if [ ! -d $binutils_version ]; then
+        wget https://ftp.gnu.org/gnu/binutils/${binutils_version}.tar.gz
+        tar -xvf ${binutils_version}.tar.gz
+    fi
+    cd ${binutils_version}
+    ./configure --prefix=${binutils_install_dir}
+    make -j$(nproc)
+    make install
+    echo "export PATH=${binutils_install_dir}/bin:\$PATH" > /etc/profile.d/binutils.sh && source /etc/profile
+    rm -rf ${binutils_version}* && cd
+}
+
 install_centos_7() {
     export PATH=/usr/local/bin:$PATH
 
-    yum -y install epel-release centos-release-scl
```

Comment:
I think we should keep it, and also update the mirror for epel repo
```
    rm /etc/yum.repos.d/CentOS-SCLo-scl.repo -f
    sed -i \
    -e 's/^mirrorlist/#mirrorlist/' \
    -e 's/^#baseurl/baseurl/' \
    -e 's/mirror\.centos\.org/vault.centos.org/' \
    /etc/yum.repos.d/CentOS-SCLo-scl-rh.repo

```

Reply:
There seems no CentOS-SCLo-scl.repo and CentOS-SCLo-scl-rh.repo files in the default centos image.
And `vault` repository does not have `epel-release` `centos-release-scl` packages.

Reply:
I mean if we keep this line to install gcc from SCL repo
`yum -y install epel-release centos-release-scl`

Reply:
It does support SCL
https://github.com/apache/arrow/pull/43127